### DISCUSSION TITLE
CLI for Custom Scripts

### DISCUSSION THREAD

**Discussion by dcramer** [remix-run/remix#2760] [Proposals]
While there might be a way to pull it off today, having a fully supported way to execute command line scripts is extremely helpful when deploying production apps. This lets you bundle in things like basic administration or other capabilities that might equate to a root-level access.

For example a common use case could be a script which creates a new user, or creates a user as an administrator.

**Comment by thmsmlr** [comment]
I'm digging around trying to find a solution for this as well. Coming from the rails community, it's really nice to have a script runner like rake to do admin tasks on my project while taking advantage of all the existing code (and polyfills) Remix provides

**Comment by thmsmlr** [comment]
For anyone stumbling across this, a work around is to use [esbuild-register](https://github.com/egoist/esbuild-register) and the `installGlobals()` function that `@remix-run/node` provides to get the fetch polyfills.

For example

```node
import { installGlobals } from '@remix-run/node';
installGlobals();

async function main() {
    console.log('do stuff');
}

main()
```

And you run the script by running

```bash
$ node -r esbuild-register path/to/script.ts
```
